crates/types/src/config/networking.rs

```diff
@@ -41,6 +41,13 @@ pub struct NetworkingOptions {
     #[serde_as(as = "serde_with::DisplayFromStr")]
     #[cfg_attr(feature = "schemars", schemars(with = "String"))]
     pub handshake_timeout: humantime::Duration,
+
+    /// # Request timeout
+    ///
+    /// Default timeout for internal cluster RPC calls.
+    #[serde_as(as = "serde_with::DisplayFromStr")]
+    #[cfg_attr(feature = "schemars", schemars(with = "String"))]
+    pub rpc_call_timeout: humantime::Duration,
```

Comment:
Unsure if this can be a single universal value. Expectations might vary a lot depending on the RPC itself. Perhaps we can define the timeout on the operation level instead?

Reply:
I read that as, it shouldn't even be configurable then? It seems like for the create-snapshot internal operation a sane value of a couple of seconds should suffice. I think it might make sense to have a general intra-cluster RPC timeout but if we don't use it now, there's no need to pollute the config keys with it yet.

Reply:
Wonder if we could integrate a default timeout in the `define_rpc!` macro, which can be optionally overridden per-operation in config. The right config section then seems more like `[worker]` since it is the "server" handling the RPC. Thoughts on this approach?

Reply:
Not sure if we should go as fine grained. Ideally, the timeout should mean something to the user and not as low-level as per-rpc type. For instance, a timeout for creating a partition store snapshot, etc.